### PR TITLE
fix: add width and height for ios devices in onCanvasResize

### PR DESCRIPTION
**Pull request by Deni32231**
Here’s a brief summary of the issues I addressed in the PR

1. Fixed incorrect chart rendering on iOS devices
2. Resolved scrolling issues on touch devices
3. Added touch end event for charts

### CODE REVIEW

File: src/lib/Chart.tsx [hunk 1]
```diff
@@ -92,7 +93,12 @@ function useCanvas(onResize: (sizeCpx: Size) => void) {
     const onCanvasResize = useCallback(
         function onCanvasResize(entries: ResizeObserverEntry[]) {
             if (canvas == null) return;
-            const { inlineSize: width, blockSize: height } = entries[0].devicePixelContentBoxSize[0];
+            const width =
+                entries[0]?.devicePixelContentBoxSize?.[0].inlineSize ??
+                entries[0].contentBoxSize[0].inlineSize * 2;
```
Comment (IlyaSkriblovsky) [inline src/lib/Chart.tsx:98]: :warning: Are you sure there should be `2`?
Shouldn't it be `devicePixelRatio`?

File: src/lib/useDragAndZoom.tsx [hunk 2]
```diff
@@ -161,7 +166,14 @@ export function useDragAndZoom(
         };
 
         const onPointerUp = (event: PointerEvent) => {
-            if (event.pointerType != "mouse") return;
+            if (event.pointerType != "mouse") {
+                const bounds = element.getBoundingClientRect();
+                if (onTouchUp) {
+                    onTouchUp(calcXInTemporaryViewport(event.pageX, bounds), event);
+                }
```
Comment (IlyaSkriblovsky) [inline src/lib/useDragAndZoom.tsx:173]: :pinching_hand:  A bit shorter :)

```suggestion
                onTouchUp?.(calcXInTemporaryViewport(event.pageX, bounds), event);
```

**Review by IlyaSkriblovsky — COMMENTED** [review]
Thanks! Please take a look on the comments.

Also, please add a couple of lines in PR description of what problem do you solve and what have been done for that.

But overall great work!

File: src/lib/Chart.tsx (hunk 1 above)
Comment (Deni32231) [reply src/lib/Chart.tsx]: Yes, you are correct! We should indeed use devicePixelRatio.

**Review by IlyaSkriblovsky — APPROVED** [review]

File: src/lib/Chart.tsx (hunk 1 above)
Comment (IlyaSkriblovsky) [reply src/lib/Chart.tsx]: :+1:

**Pull request merged**